<?xml version="1.0" ?>
<launch>
  <arg name="load_cam"          default="false"/>
  <arg name="load_bag_filepath" default="false"/>
  <arg name="bag_file"          default=""/>
  <arg name="calibration_file"  default="$(find object_keypoints_ros)/config/calibration.yaml"/>
  <arg name="image_topic"       default="/camera/color/image_row"/>
  <arg name="camera_frame"      default="camera_color_optical_frame"/>
  <arg name="model_name"        default="valve_depth"/>
  <arg name="model"             default="$(find object_keypoints_ros)/models/$(arg model_name).pt"/>
  <arg name="keypoints_file"    default="$(find object_keypoints_ros)/config/valve.json"/>
  <arg name="image_file"        default="$(find object_keypoints_ros)/resources/valve3_gt.png"/>
  <arg name="rviz"              default="true"/>
  <arg name="rviz_config"       default="$(find object_keypoints_ros)/config/valve.rviz"/>
  <arg name="gpu"               default="true"/>
  <arg name="operation_mode"    default="static"/>
  
  <node name="object_keypoints_ros" pkg="object_keypoints_ros" type="run_pipeline.py" output="screen">
    <param name="left_image_topic"    value="$(arg image_topic)" />
    <param name="right_image_topic"   value="$(arg image_topic)" />
    <param name="left_camera_frame"   value="$(arg image_topic)" />
    <param name="load_model"          value="$(arg model)"/>
    <param name="calibration"         value="$(arg calibration_file)" />
    <param name="keypoints"           value="$(arg keypoints_file)" />
    <param name="image_file"          value="$(arg image_file)"/>
    <param name="pnp"                 value="false" type="bool" />
    <param name="gpu"                 value="$(arg gpu)" type="bool" />
    <param name="verbose"             value="false"/>
    <param name="operation_mode"      value="$(arg operation_mode)"/>
  </node>

  <group if="$(arg load_cam)">
    <include file="$(find realsense2_camera)/launch/rs_camera.launch" />
  </group>
  
  <node pkg="rosbag" type="play" name="rosbag_play_recoreded" args="$(arg bag_file)" if="$(arg load_bag_filepath)" />

  <!-- Open Rviz -->
  <node name="$(anon rviz)" pkg="rviz" type="rviz" args="-d $(arg rviz_config)" output="screen" if="$(arg rviz)"/>


</launch>
